WorkflowFunctionBundle/Manager/AuthorizationWorkflowRightManager.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace OpenOrchestra\WorkflowFunctionBundle\Manager;
+
+use OpenOrchestra\WorkflowFunction\Model\WorkflowRightInterface;
+
+/**
+ * Class AuthorizationWorkflowRightManager
+ */
+class AuthorizationWorkflowRightManager
+{
+    /**
+     * Constructor
+     *
+     * @param string $authorizationClass
+     */
+    public function __construct($authorizationClass)
+    {
+        $this->authorizationClass = $authorizationClass;
+    }
+
+    /**
+     * @param array $references
+     * @param WorkflowRightInterface $workflowRight
+     *
+     * @return WorkflowRightInterface
+     */
+    public function cleanAuthorization($references, WorkflowRightInterface $workflowRight)
+    {
+        $authorizations = $workflowRight->getAuthorizations();
+        $indexAuthorizations = array();
+        foreach ($authorizations as $authorization) {
+            $id = $authorization->getReferenceId();
+            $indexAuthorizations[$id] = $authorization;
+        }
+        $authorizationsId = array_keys($indexAuthorizations);
+
+        $indexReferences = array();
+        foreach ($references as $reference) {
+            if (method_exists($reference, 'getId')) {
```

Comment:
Why do you check if the method getId exist ? if the $reference object implement ReferenceInterface, it should be here